### DISCUSSION TITLE
Add UUID v4 and v7 to edge functions

### DISCUSSION THREAD

**Discussion by fkollmann** [sqlitecloud/discussions#55] [Feature request] [answered]
Hi,

please add `uuidv4()` and `uuidv7()` to the edge functions. Using UUIDs when storing data (esp. in event streams) are very common.

This is the workaround implementation, I am currently using:

```javascript
const uuidv7 = () => { // original source: https://stackoverflow.com/a/77802638
  return 'tttttttt-tttt-7xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function(c) {
    const r = Math.trunc(Math.random() * 16);
    const v = c == 'x' ? r : (r & 0x3 | 0x8);
    return v.toString(16);
  }).replace(/^[t]{8}-[t]{4}/, function() {
    const unixtimestamp = Date.now().toString(16);
    const padded = (new Array(12 - unixtimestamp.length + 1).join('0') + unixtimestamp); // Custom padStart
    return padded.slice(0, 8) + '-' + padded.slice(8);
  });
};
```

I thinks this should be provided as an optimized version from the (SQlite Cloud) platform, instead.

Best Regards, 
Felix

**Comment by jacobprall** [comment, marked as answer]
Hey Felix! 
Thank you for the suggestion! We will add this to a list of major improvements to Edge functions planned for Q1 of 2025.

Jacob
